### ISSUE TITLE
CK1_CK2 Pad

### ISSUE THREAD

**Issue by ogamespec** [msinger/gbreveng#2]
Hi, I rebuilt the CK1_CK2 (OSC) masks and it looks like your transistor circuit and logic interpretation is different from what it actually is.

More details here:
https://github.com/emu-russia/dmgcpu/blob/main/wiki/soc/pads.md#osc

**Comment by msinger** [comment]
Oh, nice. Then I should take my schematic of this cell down. It was just a guess, based on the other I/O cells, since I haven't done its layout yet.

Did you look at the SCK pad? It also has more logic than the others. Maybe I got this wrong too.

**Comment by ogamespec** [comment]
As part of my analysis, I plan to do cross-checking with your results, everything I can get my hands on, including the rest of the pads :)

**Comment by ogamespec** [comment]
Better view of CK1/CK2 internal logic:

![CK1_CK2_Fused](https://github.com/user-attachments/assets/aef2450d-dd5b-4a21-ba72-c28a2af0241a)

![0001_lap2](https://github.com/user-attachments/assets/43e638d6-843a-4947-8509-aebd1622bd67)

**Comment by msinger** [comment]
Great, thanks. I'm wondering about the bottom part a bit more. Is this what I see?
![circuit](https://github.com/user-attachments/assets/19ba28c5-72e8-4b29-a05b-8e4e09d6d8a9)

And then the same thing basically mirrored for the ground side?
Do you know why this is? Could this be protection against voltage spikes that may be generated by the crystal? I don't know much about analog stuff. Does a crystal have characteristics of an inductor maybe? If the voltage at the pad rises and then suddenly stops, that the crystal keeps pulling it up further, way past VDD? So, it could work like the diodes. But why aren't the diodes then enough?

My thought is that if you apply a voltage at the pad, which is higher than VDD, then VDD becomes ground, and then the transistor I've drawn at the right gets conductive and pulls the voltage at the pad down to VDD.

What is your thought about these transistors in the bottom part? Did I misunderstand something? I'm not sure.

**Comment by ogamespec** [comment]
Yes it is, one of the two transistors in each of the “pieces” is “disabled” for some reason. I see this often, most likely it is just to regulate the power of the output signal (the more transistors “in parallel” - the more powerful the signal).
Analog electronics specialists will tell you more, I am not one of them :-)
